### ISSUE TITLE
Relicensing the masterlist files/repositories

### ISSUE THREAD

**Issue by Freso** [loot/loot.github.io#61]
I was thinking about our masterlists. They're really mostly objective data, which is not copyrightable data per se (which makes copyright-based licenses not apply), but due to their nature of being compiled and some parts being more subjective, I guess there is some case for them having copyright. However, I feel like we should encourage as wide use of the masterlists as possible, which the GPL somewhat hinders.

Assuming we can get permission from contributors, could we change the masterlist license(s) to [CC0](https://creativecommons.org/publicdomain/zero/1.0/) (e.g., basically releasing them into the public domain)? Or maybe [CC-by](http://creativecommons.org/licenses/by/4.0/) or [MIT](https://opensource.org/licenses/MIT)/[BSD](https://opensource.org/licenses/BSD-3-Clause)?

I realise that some of the masterlists may be a lost cause by now, esp. the ones that had a long history under BOSS and got ported to LOOT's format, but we could maybe try and reach out to the Fallout 4 masterlist contributors ([12 people](https://github.com/loot/fallout4/graphs/contributors)) and the Skyrim SE masterlist contributors ([7 people](https://github.com/loot/skyrimse/graphs/contributors))?

-----

Masterlist project issues:
- [x] [TES4: Oblivion](https://github.com/loot/oblivion/issues/24)
- [x] [TES5: Skyrim](https://github.com/loot/skyrim/issues/196)
- [x] [TES5: Skyrim SE](https://github.com/loot/skyrimse/issues/34)
- [x] [Fallout 3](https://github.com/loot/fallout3/issues/81)
- [x] [Fallout: New Vegas](https://github.com/loot/falloutnv/issues/40)
- [x] [Fallout 4](https://github.com/loot/fallout4/issues/42)

**Comment by WrinklyNinja** [comment]
> However, I feel like we should encourage as wide use of the masterlists as possible, which the GPL somewhat hinders.

I agree with encouraging wider usage, and I do acknowledge that the GPL can hinder that, but I'm not aware of any projects that could use the masterlists but don't because of licensing.

I'm OK with any of your suggested licenses, I tend to use CC0/MIT/GPL for my projects, depending on personal importance. I think for the masterlists a CC license makes more sense than a software license.

If you really wanted, you could get the contributor lists from the BOSS repositories for those masterlists that got ported, as I retained authors when I went from SVN to Git. I guess if you want to go forward with this, open an issue in each repo and @mention everyone?

**Comment by Freso** [comment]
@MaddBomber83 asks
>  Does this change let us do something we can't do now, that we want to do now?
>
> It's cool if it allows us to do more in the future, just question why now when we have nothing to balance against it.

No. This change does not allow *us*, the LOOT team, to do any more or any less than what we currently do. All(?) LOOT (related) code is GPL3, so the masterlists also being GPL3 (as they are now) has no effect on anything else in LOOT, nor would it affect any future thing "LOOT" would want to do with them, as that would still most likely be "GPL v3 (or later)", and thus same license as the masterlists.

What the change *will* do is make the masterlists available under an infinitely more free license so people like Mator (author of Merge Plugins, Mator Smash, Mod Picker, etc.), the xEdit team (or anyone making xEdit plugins), university students or researchers, etc. can be sure they can use the LOOT masterlists/masterlist data in their projects, regardless of what license they choose to publish those projects under.

**Comment by Freso** [comment]
I've made issues for all the masterlist projects now:
- [ ] [TES4: Oblivion](https://github.com/loot/oblivion/issues/24)
- [ ] [TES5: Skyrim](https://github.com/loot/skyrim/issues/1960)
- [x] [TES5: Skyrim SE](https://github.com/loot/skyrimse/issues/34)
- [ ] [Fallout 3](https://github.com/loot/fallout3/issues/81)
- [ ] [Fallout: New Vegas](https://github.com/loot/falloutnv/issues/40)
- [x] [Fallout 4](https://github.com/loot/fallout4/issues/42)

**Comment by Freso** [comment]
Of the e-mail addresses dug up by @WrinklyNinja of BOSS contributors not found on GitHub, 5 e-mail addresses seem to no longer exist (`<marklonn at gmail.com>`, `<wyll at live.de>`, `<jpearce at pacific.net.au>`, `<simon888 at cox.net>`, and `<venkukad at gmx.de>`) and one doesn't have an address specified ("deaths_soul (MCP)"). That means there are at least 6 contributors we won't be able to reach out to.

This leaves us a couple of options:
1. Try and find them by other means. It's unlikely that we will find all of them though, but I guess we can try.
2. Dig into the BOSS commit logs and see exactly which contributions they have made and if any remnant of these are still in the masterlists, and if so, remove them (and have them remade, if warranted).
3. Move ahead anyway. 6 people, all pre-dating LOOT, are unlikely to have had a "significant" level of contribution to the current state of affairs, and the copyright status of the masterlists is dubious at best to begin with, with the masterlists being akin to a database, and not all jurisdictions have a "sweat of the brow"/database doctrine, and even if so, any single, individual contribution is unlikely to have enough impact to warrant copyright of its own (copyright requires e.g., certain levels of originality). I'm not really happy with this approach and would prefer approaches 1. or 2. before this one, if possible. Even if it might be legally defensible, I'd rather have everyone on board rather than someone not getting heard.
4. Not changing the license at all.
5. Something else?

**Comment by WrinklyNinja** [comment]
[This authors mapping](https://github.com/boss-developers/boss-developers.github.io/wiki/Moving-from-SVN) I'd forgotten I wrote shows:

- marklonn is Malonn (still active on Nexus)
- wyll is @noxwyll
- simon888 is Psymon (I don't know if he's still active anywhere)

**Comment by noxwyll** [comment]
@WrinklyNinja Am active on nexus in fallout 4 and skyrim. Skyrim (Gothic Project) and Fallout 4 (Weapon Mod Project). Anyways my actual nexus name is iyumichan, I organized and for the majority of the time lead the Mission Mojave project.

**Comment by Freso** [comment]
I'm guessing that deaths_soul is @deathssoul and venkukad is @KadVenku?

**Comment by Freso** [comment]
@TokcDK confirmed that menduki is a previous e-mail alias of theirs here: https://github.com/loot/skyrim/issues/196#issuecomment-410246782

**Comment by KadVenku** [comment]
@Freso You are correct, that's an old e-mail address of mine.
In any case, you have full permission to do whatever you deem necessary with my BOSS contributions.

**Comment by Vacuity729** [comment]
@Freso You have my permission to make the license changes.  If the masterlists become more useful that way, I'm all for it.  I'll send you an email in a sec to confirm that I'm not a random person pretending to be an old contributor.
@WrinklyNinja Yo!  Long time, no see!  I hope you're doing well!

**Comment by spaceoden** [comment]
All for it. How the client developers license their work is their business, but the masterlist was always a public project. I can think of no reason to restrict it.

**Comment by Ortham** [comment]
Considering that the list of contributors is probably still growing and that some existing contibutors are probably not contactable (and so we'll probably never be able to relicense some masterlists), does anyone know if it's possible to use a new license for only new contributions, and document that contributions made before YYYY-MM-DD are licensed under the GPL, and contributions made after that date are licensed under CC0? 

My thinking is that if new contributors already agree to CC0 as part of contributing, we can't end up missing anyone when trying to get all the necessary agreements.

Also, at some point it may be worth finding out how extensive the contributions by those who have not yet agreed are, and whether they're still present in the latest masterlists. Presumably, if a contributor's contributions are no longer part of the masterlist then their agreement is not required to relicense the current version (but obviously past versions containing their work would still be GPL'ed). IANAL though.

**Comment by sibir-ine** [comment]
> My thinking is that if new contributors already agree to CC0 as part of contributing, we can't end up missing anyone when trying to get all the necessary agreements.

Yeah, I think, going forward, it'd be for the best that we ask new contributors to take a look at this issue before we can merge their changes.

> Also, at some point it may be worth finding out how extensive the contributions by those who have not yet agreed are, and whether they're still present in the latest masterlists.

I think this is inevitable. I want to try to pick up the issue from where @Freso last left it & try to reach out to whoever I can, however I can. If, after that & some time (maybe a few months), there are still some people left, I'd personally rather we go forward with relicensing, even if it means removing contributions from contributors that haven't responded/agreed. We should definitely check to see if any of them don't have any changes that are still in the masterlist, though. That'd save us some time, assuming we don't still need their consent. Though, we probably should contact a lawyer or at least do some research about it just in case. 🤷 I'll see if anyone I know knows any.

**Comment by NickDickinsonWilde** [comment]
@sibir-ine got your dm, thanks for reaching out. I'm 100% good with re-licensing any of my prior contributions to the BOSS/LOOT/etc load ordering lists to CC0. Thanks also for taking on maintainership!

**Comment by sibir-ine** [comment]
@NickDickinsonWilde Thanks for responding! & of course, it's a project I feel is invaluable for making modding easier & more accessible to people around the world, so I'm happy I can be a part of keeping that alive & well!

**Comment by sibir-ine** [comment]
Just realized that the Contributors tab in Insights doesn't list people whose GitHub accounts no longer exist, so we've probably missed some people (confirmed for FNV). We'll have to use `git shortlog` for the remaining LOOT masterlists to double-check. @Ortham, I'm assuming that's what you did to get the list of BOSS contributors, right? If so, this just applies to the LOOT masterlists.

**Comment by sibir-ine** [comment]
There's also the issue of the older branches. Since most of the people that probably won't respond will be from BOSS, they're contributions will be in all the old branches. Since they're seldom used & aren't actively maintained, I think it'd be for the best if we just removed their contributions from those branches as opposed to carrying over our clean-room reimplementations to them. I don't think it's worth the effort converting our reimplementations between the different syntaxes.

**Comment by Ortham** [comment]
> @Ortham, I'm assuming that's what you did to get the list of BOSS contributors, right? If so, this just applies to the LOOT masterlists.

Yes, that's what I did for BOSS.

**Comment by sibir-ine** [comment]
> Relicensing will only affect future versions of the masterlist

Ah, I thought it'd affect all branches. Yeah, unless someone can come up with a good reason to relicense the older branches as well, let's just leave them.

**Comment by silentdark** [comment]
You have my permission, masterlist is not software, you can change the license to CC0.

**Comment by sibir-ine** [comment]
Thank you so much for responding!

**Comment by sibir-ine** [comment]
Ok, added everyone that was missing along with their GitHub accounts if I could find them.

**Comment by sibir-ine** [comment]
I just realized another issue: what about people who've provided translations to the non-CC0 masterlists that we committed on their behalf? For stuff like users pointing out or providing metadata like cleaning data or load rules that we then add, I think it's fair to say that that isn't their copyright & we therefore don't need their agreement. However, for things like translations, that seems like a much greyer area. For translations that are on CC0 & non-CC0 masterlists, they practically released them into the public domain by allowing them to be on the CC0 masterlists, so they've implicitly agreed. However, for translations exclusive to non-CC0 masterlists, I'm not sure we're in the clear. We might be able to argue that since we added the translations, there was an implicit understanding that they'd be released into the public domain before we'd be willing to add them. However, I'm not confidant in this argument unless there was explicit communication that the translator would have to release their translations into the public domain first. Therefore, it may be for the best that we ask those translators just in case. If I'm correct & we indeed have to, that unfortunately means that for translators who don't respond, we'd have to remove their translations entirely.

**Comment by Ortham** [comment]
I think that would be going a bit too far and creating a problem where none exists in practice. I question if the translations qualify as something copyrightable, but even if we assume they are, very few translators are committing to the masterlist repositories themselves, putting the others in the same position as the innumerable people who have provided the LOOT team with information to put into the masterlists. Are we going to block relicensing on the impossible task of chasing all of them down? If not, what's the difference, where do we draw the line and why?

**Comment by TsunaMoo** [comment]
As a previous Localization Specialist I can tell you that translations are definitely under copyright.
Different question tho, when people contribute do they agree to any form of agreement?
If not that might at least be a good idea for the future.

And honestly this is a collection of data, can that even be under copyright? 
Wouldn't the mechanics itself be the copyrighted part?
And would anyone have contributed to this with the expectation that they hold the copyright to their segment, or was it more done as a service to the community, aka possibly public domain?

**Comment by sibir-ine** [comment]
> I think that would be going a bit too far and creating a problem where none exists in practice. I question if the translations qualify as something copyrightable, but even if we assume they are, very few translators are committing to the masterlist repositories themselves (and we're already seeking their agreement), putting the others in the same position as the innumerable people who have provided the LOOT team with information to put into the masterlists. Are we going to block relicensing on the impossible task of chasing all of them down? If not, what's the difference, where do we draw the line and why?

For people who provide us with information, I think that situation is different. Since we commit the information, the specific implementation & therefore copyright comes down to us. If they provided something like a specific message that we copied verbatim, then I would assume that'd be their copyrighted material that they provided to us that falls under the license of the specific masterlist it was provided to, so not necessarily public domain unless otherwise implied or specified somehow. I'm fairly certain translations fall under the same standard that messages in general do as they're just as subjective, & we've been erring on the side of copyright on that issue so far. It might be enough to say that before we committed their translations, there has been an understanding these past 4 years since this issue was filed that they were releasing it into the public domain first. My thinking on this isn't concrete yet, it's just something I think we should consider.

**Comment by Ortham** [comment]
> As a previous Localization Specialist I can tell you that translations are definitely under copyright.

I understand that being the case for significant chunks of text, but if I asked you what the word for "error" is in German and you replied with "Fehler", how is that any different from me just looking it up in an English-German dictionary? There's no originality or creativity in what you'd have done in answering me, you'd have effectively just read from the English-German dictionary you keep in your own memory and passed that information on to me.

> Different question tho, when people contribute do they agree to any form of agreement?

No, there's not a formal process. It's something I brought up recently, but I think relicensing under CC0 would make having a contributor agreement pointless.

> And honestly this is a collection of data, can that even be under copyright?

Maybe?

> Wouldn't the mechanics itself be the copyrighted part?

Yes, but not necessarily exclusively so.

> And would anyone have contributed to this with the expectation that they hold the copyright to their segment, or was it more done as a service to the community, aka possibly public domain?

I think it's almost certain that everyone who contributes doesn't really care about the licensing, but I can't assume that's the case or know whether it is or not without asking.

**Comment by TsunaMoo** [comment]
No, stuff that's just single words from dictionary is surely not copyrighted, also copy paste stuff where you just swap subject and object text also unlikely are under copyright.

**Comment by Ortham** [comment]
> No, stuff that's just single words from dictionary is surely not copyrighted, also copy paste stuff where you just swap subject and object text also unlikely are under copyright.

OK, thanks for clarifying.

**Comment by TsunaMoo** [comment]
It should be sufficient to make a request to change license, give time for people to say no, if that doesn't happen after a month, should just proceed. Then you gave the time for objections which should be reasonable and sufficient.

The main reason for translation copyright is because your own creativity and interpretation flows into it.

**Comment by sibir-ine** [comment]
> I understand that being the case for significant chunks of text, but if I asked you what the word for "error" is in German and you replied with "Fehler", how is that any different from me just looking it up in an English-German dictionary?

That's the gray area. If we only asked them to translate a single word, we'd be able to argue that that isn't their copyright. But we ask them to translate phrases & sentences, & there are definitely different & unique ways to translate them that may pass for originality.

**Comment by sibir-ine** [comment]
> It should be sufficient to make a request to change license, give time for people to say no, if that doesn't happen after a month, should just proceed. Then you gave the time for objections which should be reasonable and sufficient.

Unfortunately, we can't assume everyone will have seen it & silently agreed if they don't respond. We will proceed regardless (probably in a few months), but we'll just have to reimplement or remove the contributions of people who never responded.

**Comment by TsunaMoo** [comment]
> 
> Unfortunately, we can't assume everyone will have seen it & silently agreed if they don't respond. We will proceed regardless (probably in a few months), but we'll just have to reimplement or remove the contributions of people who never responded.

Yes but there is a legal example for this stuff, as long as this request is easy to view and public this plan should be able to proceed as long as no objections happen because it can be assumed that everyone had at least a chance to view it.

**Comment by sibir-ine** [comment]
> Yes but there is a legal example for this stuff, as long as this request is easy to view and public this plan should be able to proceed as long as no objections happen because it can be assumed that everyone had at least a chance to view it.

That could possibly be the case. However, none of us are lawyers, so we opted to err on the side of caution.

**Comment by sibir-ine** [comment]
@Ortham Okay, I talked to a family member who works in a business that develops open source software, & they have someone well versed in open source licenses & copyright that would be happy to answer some questions. If he doesn't know, they could easily contact a lawyer too.

We should ask him if TsunaMoo is correct that as long as everyone had a reasonable chance to view this issue, that we can proceed without everyone's agreement & without reimplementing anything.

We should also ask him about about translations that we committed on behalf of a translator & whether that can be considered copyright.

Are there any other questions you have or anyone else has?

**Comment by sibir-ine** [comment]
@Ortham @Freso @MacSplody @pStyl3 @Infernio 

Okay, I asked him a couple of questions, & he clarified some points of confusion I had. If anyone's curious, he's the founder of [EJBCA](https://www.ejbca.org/), so he's had a lot of experience & legal counsel on copyright as it's applied to open-source projects. I'll summarize the key points he made below.

First, I'd like to just cement the deadline for officially beginning the process of relicensing the remaining masterlists. I like pStyl3's suggestion of July 6th, 2021, the 4th anniversary of this issue (though I'd prefer July 1st just to round away those extra 5 days 😛 ). Four years is plenty for contributors to make meaningful arguments against the license change, & no one has even objected so far. Also, for those we've reached out to who haven't responded so far, I think it's safe to say they probably never will, so we probably won't get anymore people to explicitly agree if we continue to wait. We also continue to get new contributors, so it just adds to the people we have to ask. I think we should set the deadline as July 1st, 2021. If anyone else has a better date & a reason why, feel free to comment.

Now, for the summary:
- This is more for confirmation, but we indeed can't take no-responses as silent agreements. We'll have to either reimplement or remove.
- Also for confirmation, but, for the purposes of this project, it's probably safe to say that translations that we committed on a translators behalf do not need to be removed/reimplemented.
- Also, this is a discussion we can have in more detail once the deadline has passed, but, considering some of the masterlists will require extensive work to relicense (see https://github.com/loot/skyrim/issues/196 & its outstanding contributors' [contributions](https://github.com/loot/skyrim/issues/196#issuecomment-817593565)), I think we should consider taking a more lenient approach to what needs to be reimplemented. I'm all for erring on the side of caution, but maybe we should eventually think about what pieces of metadata are safe to say don't pass the threshold for originality (e.g., maybe Bash Tags, cleaning data, translations). He argued that the masterlists are probably mostly not copyrightable, so we can probably take some liberties to at least somewhat reduce the absurd workload ahead of us. I'll still do as much as I can even if we decide not to take any liberties, but I think it's still worth considering.
- Finally, I saw this [discussed](https://github.com/ModOrganizer2/modorganizer/issues/1106#issuecomment-638973777) by the MO2 devs & asked about it, but maybe we could get some use out of having a contributor license agreement. It's mostly too-little-too-late for the purposes of this issue, but it could still be useful if we end up wanting to make another major change that requires us to reach out to every contributor. Nothing comes to mind, but it's worth at least considering. At the very least, for this issue, it'd make it easier to deal with new contributors' contributions until we've relicensed every masterlist. That way, we wouldn't need to worry about asking them to look at the relevant issue & state whether they agree. It'd probably be more useful for the more code-heavy repositories, but that's a discussion for another issue.

**Comment by Superfly-Johnson** [comment]
I approve. It's no big deal and if it prevents doing any rollbacks then so be it.

I would like to add that I agree with the fact that metadata shouldn't be licensed.

**Comment by sibir-ine** [comment]
@Superfly-Johnson Thanks! I just want to double-check, did you contribute to one of the GPLv3 masterlists under a different name, or are you agreeing for any future contributions you may make to them?

**Comment by pStyl3** [comment]
@Superfly-Johnson contributed to the Fallout 4 masterlist, see [here](https://github.com/loot/fallout4/commits?author=Superfly-Johnson&since=2021-01-31&until=2021-02-28), which is already under the CC0 lincese.

**Comment by pStyl3** [comment]
@RacerBG has agreed on Discord.

> I don't know if translators are involved in the relicensing process but if my permission is needed - I fully agree, you can do whatever you want.

**Comment by pStyl3** [comment]
@IllusiveMan196 has agreed on Discord. (Russian and Ukrainian translations)

> no objections here

**Comment by Allen1351** [comment]
As the translator of the Spanish version, I agree. Keep the good work.

**Comment by pStyl3** [comment]
# 1st of July 2021

As written above, and in Discord:
> While our masterlists for Skyrim Special Edition and Fallout 4 have already been changed to CC0, and the masterlists for Morrowind and Enderal have been created under the CC0 license, we still need to change the license for our Oblivion, Skyrim, Fallout 3, and Fallout NV masterlists.
> 
> This process has been going on for almost 4 years by now, partly because we want the approval of everyone that contributed to the individual masterlists, so we have decided to set a deadline for everyone from which we haven't heard back from until now. This date will be the **1st of July 2021** until which every contributor has time to agree or disagree regarding our intentions, may it be via GitHub, Email, Private Message on Nexus or AFKMods (where our LOOT Thread is hosted), or here on Discord.
> 
> Even after this date, people who have contributed will of course be able to state their opinions with the caveat that we will start to work towards relicensing the remaining masterlists regardless.
> 
> In detail, this means that we will either **rework** or **remove** contributions from the people that we weren't able to get in contact with and, as such, we weren't able to get their agreement. This, of course, would only be the last possible course of action for us; ideally, we'd like to get the approval of every contributor.

The approval of these remaining contributors is missing:

### Skyrim:
@kuroko137
@ArkangelCorvus
@v111v111
Torrello `<torrello.boss at gmail.com>` ([Nexus](https://www.nexusmods.com/users/2018867))
Calen Ellefson `<tigercrab505 at gmail.com>` ([Nexus](https://www.nexusmods.com/users/3621086))

### Oblivion:
gabrielmonn `<gabrielmonn at gmail.com>`

### Fallout 3:
Torrello `<torrello.boss at gmail.com>` ([Nexus](https://www.nexusmods.com/users/2018867))
maahaa12 `<maahaa12 at gmail.com>`

### Fallout NV:
MothHeart `<kristianityofficial+4 at gmail.com>`
Skyline `<mariolam9462 at gmail.com>`

**Comment by Infernio** [comment]
The gabrielmonn reimplementations [are almost done](https://gist.github.com/Infernio/d1c02e4436cb1c96d5a85f33bf3891a9). I won't have time to finish them off for a while though (because WB is just about to reach Python 3).

**Comment by pStyl3** [comment]
The deadline has been reached. We now will actively start the process of relicensing the masterlists. As laid out above, the remaining contributors from which we haven't heard back will of course continue to be able to voice their opinion, but we will start reworking or removing their contributions now, so that the **Oblivion**, **Skyrim**, **Fallout 3** and **Fallout NV** masterlists can be relicensed under CC0 too.

Thanks to everyone who agreed! 👍

**Comment by sibir-ine** [comment]
@Infernio has Oblivion covered, so I'll finish the to-do lists for New Vegas & FO3 to start working on them. Skyrim will be a whole other beast entirely, especially all of the Japanese translations. For those, we'll want to finish https://github.com/loot/skyrim/issues/452 first to reduce & centralize the translations as much as possible, then see if another Japanese translator would be willing to reimplement the remainder.

**Comment by Infernio** [comment]
I'll finish off the last couple Oblivion reimplementations tomorrow 👍

**Comment by sibir-ine** [comment]
After four years, this issue can finally be closed! Thank you so much to everyone who agreed & helped in the relicensing process!
